include/painless/parameter.h

```diff
@@ -141,9 +149,7 @@ class Parameter {
 
  private:
   std::string getFilename() const {
-    std::string filename = BASE_PATH;
-    filename += m_parameter_name;
-    return filename;
+    return m_base_path + m_parameter_name;
   }
```

Comment:
There are two to three allocations (depending on NRVO) in this function.  That is at least one too many.

Reply:
Before your change, both `BASE_PATH` and `m_parameter_name` were `const char*`. How would you write that with a single allocation (let's assume RVO)? With `.reserve(PATH_MAX)`? By computing the length first (that seems inefficient)?

Reply:
You can actually use a (thread local) static buffer of size `PATH_MAX` for that.  But as long as `getFilename` returns a `std::string` you won't get around the allocation.

Reply:
Something like this:
```cpp
inline const char *get_base_path() {
  static thread_local char buffer[_POSIX_PATH_MAX];
  const char *tmpdir = nullptr;
  if ((tmpdir = getenv("TMPDIR")) == nullptr) {
      tmpdir = "/tmp";
  }
  char *pos = buffer;
  while (pos - buffer < _POSIX_PATH_MAX && (*pos = *tmpdir++) != '\0') {
      ++pos;
  }
  buffer[_POSIX_PATH_MAX - 1] = '\0';
  return buffer;
}
```

Reply:
I think I would go with the straightforward-but-inefficient code for now, as we do not have to call `get_base_path()` repeatedly.